ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py

```diff
@@ -646,6 +646,37 @@ def test_nanmedian(
     )
 
 
+@handle_test(
+    fn_tree="functional.ivy.experimental.nanmin",
+    dtype_x_axis_castable=_get_castable_float_dtype_nan(),
+    test_gradients=st.just(False),
+    keep_dims=st.booleans(),
+)
+def test_nanmin(
+    *,
+    dtype_x_axis_castable,
+    keep_dims,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+):
+    input_dtype, x, axis, castable_dtype = dtype_x_axis_castable
+    x = x[0]
+    helpers.test_function(
+        input_dtypes=[input_dtype],
+        test_flags=test_flags,
+        rtol_=1e-1,
+        atol_=1e-1,
+        backend_to_test=backend_fw,
+        fn_name=fn_name,
+        on_device=on_device,
+        a=x,
+        axis=axis,
+        keepdims=keep_dims,
```

Comment:
we should be generating values for the `initial` and `where` arguments as well in the tests

Reply:
I have generated initial values for initial but I am not sure of "where" as in the file
there are functions which take "where" arguments but the tests are not generated for it and nor there is any example in the backends of handling where.

Reply:
Basically we should be generating the `where` argument in those tests too. In general, every argument accepted by the function (except the `out` argument) should be generated in the data generation in the tests as well. For the `out` argument, it gets created inside the `test_function` anyway so we don't need to generate the `out` argument but we should generate every other argument.

Reply:
> Basically we should be generating the `where` argument in those tests too. In general, every argument accepted by the function (except the `out` argument) should be generated in the data generation in the tests as well. For the `out` argument, it gets created inside the `test_function` anyway so we don't need to generate the `out` argument but we should generate every other argument.

@vedpatwardhan Can you give me an idea how should I generate the values of `where` argument and how to pass the test cases of it across the backends. Because I tried, but I am not sure how where argument actually works by reading the documentation.

Reply:
@Aaryan562 you can see the `where` argument in more detail [here](https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduce.html#numpy.ufunc.reduce)

Reply:
> @vedpatwardhan Can you give me an idea how should I generate the values of `where` argument and how to pass the test cases of it across the backends. Because I tried, but I am not sure how where argument actually works by reading the documentation. Also, there are no examples of `where` argument in the files through which I can take motivation

Basically the value of the `where` argument should be a boolean array broadcastable to `x`, there's helpers like `broadcastable_arrays` which could be used. @HaiderSultanArc could help out further 😄

Reply:
@vedpatwardhan I searched in the docs but I did not find any helpers like `broadcastable_arrays` there are `array_and_broadcastable_shape` and `array_bools`

Reply:
`broadcastable_arrays` is present in `test_core/test_dtype.py`, `mutually_broadcastable_shapes` is present in `helpers/hypothesis_helpers/array_helpers.py`. Not sure why they don't show up in the docs